### PR TITLE
Read RekorUrl and PublicKey from ECP

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-767

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

File: internal/policy/policy.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	ecc "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	cosignSig "github.com/sigstore/cosign/pkg/signature"
+	sigstoreSig "github.com/sigstore/sigstore/pkg/signature"
+	log "github.com/sirupsen/logrus"
+)
+
+// WithPublicKey returns a new instance of EnterpriseContractPolicy if an update
```
Comment (joejstuart) [inline internal/policy/policy.go:33]: typo. `WithRekorUrl`

Comment (lcarva) [reply internal/policy/policy.go]: Guilty of copy and paste! Fixed :)

File: internal/kubernetes/client.go [hunk 2]
```diff
@@ -66,20 +65,37 @@ func createControllerRuntimeClient() (client.Client, error) {
...
-	if err != nil {
-		log.Debug("Failed to get policy from cluster!")
+	if err := k.client.Get(ctx, *name, policy); err != nil {
+		log.Debugf("Failed to fetch the policy from cluster: %s", err)
 		return nil, err
 	}
-	policyJson, _ := json.Marshal(policy.Spec)
-	log.Debugf("Policy fetched:\n%s", policyJson)
+	if policyJson, err := json.Marshal(policy.Spec); err != nil {
+		log.Debugf("Unable to marshal policy to json: %s", policyJson)
```
Comment (joejstuart) [inline internal/kubernetes/client.go:94]: Do you need to return the err after logging?

File: internal/policy/policy.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	cosignSig "github.com/sigstore/cosign/pkg/signature"
+	sigstoreSig "github.com/sigstore/sigstore/pkg/signature"
+	log "github.com/sirupsen/logrus"
+)
+
+// WithRekorUrl returns a new instance of EnterpriseContractPolicy if an update
+// to its rekorUrl is needed. Otherwise, the same instance is returned.
+func WithRekorUrl(policy *ecc.EnterpriseContractPolicy, rekorUrl string) *ecc.EnterpriseContractPolicy {
```
Comment (joejstuart) [inline internal/policy/policy.go:35]: What if the rekorUrl is passed on the cli and it's in the policy? Is the user notified the cli param is ignored? Or should the policy be overridden?

File: internal/kubernetes/client.go (hunk 2 above)
Comment (lcarva) [reply internal/kubernetes/client.go:94]: Well I don't know... The call to `json.Marshall` is only meant to assist in logging (which seems a bit odd to me). In the previous code, the error code was ignored completely. My assumption is that it was intended to be an optional thing. I can't picture a scenario where the `json.Marshall` call actually fails though. I can switch back to ignoring the error as it was before?

File: internal/policy/policy.go (hunk 3 above)
Comment (lcarva) [reply internal/policy/policy.go:35]: I had a [discussion|https://issues.redhat.com/browse/HACBS-767?focusedCommentId=20778298&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20778298] with myself about the precedence of these values. I reasoned that parameters from the CLI should overwrite whatever is in the ECP.

In `cmd/validate_image.go`, I changed the description of the rekorUrl and publicKey parameters to indicate this behavior. Does that address your concern?

File: internal/kubernetes/client.go (hunk 2 above)
Comment (joejstuart) [reply internal/kubernetes/client.go:94]: you're right, I didn't look close enough. Maybe log `err` though instead of `policyJson`?

Comment (lcarva) [reply internal/kubernetes/client.go]: I took it a step further. I removed the `json.Marshall` call altogether. Instead, I just log the policy object directly. I'm using the `%#v` "verb" now so it actually shows up in a more readable way. Wdyt?

File: internal/policy/policy.go (hunk 3 above)
Comment (joejstuart) [reply internal/policy/policy.go:35]: perfect!

**Comment by joejstuart** [comment]
LGTM

**Pull request merged**